public/javascripts/janus.js

```diff
@@ -279,4 +279,33 @@ jQuery(function($){
         }
     });
 
+    // auto-logout switch (persisted with local storage)
+    const LOCAL_STORAGE_KEY__AUTO_LOGOUT = "autoLogout"
```

Comment:
I think this will break on pages other than those that include the logout switch.

The approach the other "modules" take is to put the logic for each piece of functionality behind a check that this module is included on the current page, which has two benefits.

Firstly, this means that each piece of functionality works wherever it is included, without having to explicitly list the routes where it would apply. For example, the account picking functionality is included on the homepage, but also appears on the support and admin pages for people on support and people with admin access respectively. Rather than duplicate the logic for deciding whether the account picking logic should apply, this JS file is included everywhere and the setup for each module happens when its elements are found.
In this case we'd want to have a think about whether all of these account-picking pages should include the auto-logout behaviour, and if so, make sure that the element it is tied to appears in all these cases.

Secondly, this approach isolates each module in its own closure. Hopefully putting things behind the element check means everything always works, but if there are problems then errors in each closure won't break the rest of the page, so we only lose one module at a time.
In this case, pages that don't include this auto logout switch element will fail when we try and add properties and events to the element. This block will fail, and because it isn't isolated, it will stop executing this whole codeblock (which means potentially breaking all the modules on the page).

Reply:
Actually, I thought I'd check (it has been a while) and I think this will be fine in practice - it'll be ok trying to do things to an empty element selection. I'll leave this comment as a description of why things are laid out the way they are though, you might still choose to put the setup behind a check that the `#auto_login_switch` element exists for consistency but up to you!